lib/jekyll/polyglot/patches/jekyll/site.rb

```diff
@@ -125,14 +132,26 @@ def relative_url_regex
       @exclude.each do |x|
         regex += "(?!#{x}\/)"
       end
-      url_quoted = config['url']
-      url_quoted = Regexp.quote(url_quoted) unless url_quoted.nil?
-      %r{href=\"(?:#{url_quoted})?#{@baseurl}\/((?:#{regex}[^,'\"\s\/?\.#]+\.?)*(?:\/[^\]\[\)\(\"\'\s]*)?)\"}
+      %r{href=\"?#{@baseurl}\/((?:#{regex}[^,'\"\s\/?\.#]+\.?)*(?:\/[^\]\[\)\(\"\'\s]*)?)\"}
```

Comment:
It should be intentional. The exclusion of the `/:lang` from absolute urls is needed to ensure that the i18n_headers are mostly left alone.

You are right thought a relative url like `/en/about`  will be processed into `/en/en/about` which isn't right. Maybe the relative urls should get the same treatment.